slides/k8s/build-with-docker.md

```diff
@@ -0,0 +1,156 @@
+# Building images with the Docker Engine
+
+- Until now, we have built our images manually, directly on a node
+
+- We are going to show how to build images from within the cluster
+
+  (by executing code in a container controlled by Kubernetes)
+
+- We are going to use the Docker Engine for that purpose
+
+- To access the Docker Engine, we will mount the Docker socket in our container
+
+- After building the image, we will push it to our self-hosted registry
+
+---
+
+## Resource specification for our builder pod
+
+.small[
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: build-image
+spec:
+  restartPolicy: OnFailure
+  containers:
+  - name: docker-build
+    image: docker
+    env:
+    - name: REGISTRY_PORT
+      value: "`3XXXX`"
+    command: ["sh", "-c"]
+    args:
+    - |
+      apk add --no-cache git &&
+      mkdir /workspace &&
+      git clone https://github.com/jpetazzo/container.training /workspace &&
+      docker build -t localhost:$REGISTRY_PORT/worker /workspace/dockercoins/worker &&
+      docker push localhost:$REGISTRY_PORT/worker
+    volumeMounts:
+    - name: docker-socket
+      mountPath: /var/run/docker.sock
+  volumes:
+  - name: docker-socket
+    hostPath:
+      path: /var/run/docker.sock
+```
+]
+
+---
+
+## Breaking down the pod specification (1/2)
+
+- `restartPolicy: OnFailure` prevents the build from running in an infinite lopo
+
+- We use the `docker` image (so that the `docker` CLI is available)
+
+- We rely on the fact that the `docker` image is based on `alpine`
+
+  (which is why we use `apk` to install `git`)
+
+- The port for the registry is passed through an environment variable
+
+  (this avoids to repeat it in the specification, which would be error-prone)
+
+.warning[The environment variable has to be a string, so the `"` are mandatory!]
```

Comment:
`"`s would read more clearly in English.

Reply:
:+1: